src/jdk.jartool/share/classes/sun/security/tools/jarsigner/Main.java

```diff
@@ -1054,7 +1063,7 @@ private void displayMessagesAndResult(boolean isSigning) {
                 notYetValidCert || chainNotValidated || hasExpiredCert ||
                 hasUnsignedEntry || signerSelfSigned || (legacyAlg != 0) ||
                 (disabledAlg != 0) || aliasNotInStore || notSignedByAlias ||
-                tsaChainNotValidated ||
+                tsaChainNotValidated || permsDetected ||
```

Comment:
I'd prefer move the `permsDetected` check and the if block for it out of this big block, since it's not a fatal warning and just informational. You can move it to the big if block of `if (hasExpiringCert...`.